synapse/lib/agenda.py

```diff
@@ -863,19 +870,24 @@ async def _runJob(self, user, appt):
                     # we only care about the last five errors
                     'lasterrs': list(appt.lasterrs[-5:]),
                 }
-                await self.core.addCronEdits(appt.iden, edits)
+
+                if self.core.isactive:
+                    await self.core.addCronEdits(appt.iden, edits)
 
             took = finishtime - starttime
             mesg = f'Agenda completed query for iden={appt.iden} name={appt.name} with result "{result}" ' \
                    f'took {took:.3f}s'
+            if not self.core.isactive:
+                mesg = mesg + ' Agenda status will not be saved since the Cortex is no longer active.'
```

Comment:
Minor nit on the message text to maybe reduce some confusion.
```suggestion
                mesg = mesg + ' Agenda status will not be saved since the Cortex is no longer the leader.'
```